### PR TITLE
[SFEQS-1222] Keep Signature Request statuses consistent across the system

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

1. Add two functions on the issuer side `MarkAsSigned` and `MarkAsRejected`
2. Add event emitters (backed by Azure Storage Queue)

#### Motivation and Context

since the signature request is stored in two separate databases, we have to keep them in sync

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-func-sign-issuer/src/app/use-cases/mark-request-rejected.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ) =>
+  (request: SignatureRequestRejected) =>
+    pipe(
+      pipe(request.issuerId, getSignatureRequest(request.id)),
+      TE.chain(
+        TE.fromOption(
+          () => new EntityNotFoundError("Signature Request not found.")
+        )
+      ),
+      TE.chainEitherK(markAsRejected(request.rejectedAt, request.rejectReason)),
```
Comment (gunzip) [inline apps/io-func-sign-issuer/src/app/use-cases/mark-request-rejected.ts:26]: what if the enque operation fails?

File: apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  ) =>
+  (request: SignatureRequestSigned) =>
+    pipe(
+      pipe(request.issuerId, getSignatureRequest(request.id)),
+      TE.chain(
+        TE.fromOption(
+          () => new EntityNotFoundError("Signature Request not found.")
+        )
+      ),
+      TE.chainEitherK(markAsSigned),
```
Comment (gunzip) [inline apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts:26]: ditto

File: apps/io-func-sign-user/src/app/use-cases/validate-signature.ts [hunk 3]
```diff
@@ -55,6 +62,9 @@ export const makeMarkSignatureAndSignatureRequestAsRejected =
           markAsRejected(rejectedReason),
           TE.fromEither,
           TE.chain(upsertSignatureRequest),
+          TE.chainFirst((r) =>
```
Comment (gunzip) [inline apps/io-func-sign-user/src/app/use-cases/validate-signature.ts:65]: needs unit tests

File: apps/io-func-sign-user/src/app/use-cases/validate-signature.ts [hunk 4]
```diff
@@ -133,11 +146,17 @@ export const makeValidateSignature =
...
                   ),
                   A.sequence(TE.ApplicativeSeq),
+
                   TE.map((documents) => ({
                     ...signatureRequest,
                     documents,
                   })),
                   TE.chainEitherK(markAsSigned),
+                  TE.chainFirst((r: SignatureRequest) =>
+                    notifySignatureRequestSignedEvent(
```
Comment (gunzip) [inline apps/io-func-sign-user/src/app/use-cases/validate-signature.ts:156]: ditto

File: apps/io-func-sign-user/src/app/use-cases/validate-signature.ts (hunk 3 above)
Comment (lucacavallaro) [reply apps/io-func-sign-user/src/app/use-cases/validate-signature.ts:65]: this use case is part of a big refactor that I'm developing in parallel and will be released with one of the next PRs

it has been rewritten to make it more linear and have less responsibility, so unit tests will come with that change

File: apps/io-func-sign-issuer/src/app/use-cases/mark-request-rejected.ts (hunk 1 above)
Comment (lucacavallaro) [reply apps/io-func-sign-issuer/src/app/use-cases/mark-request-rejected.ts:26]: if an error occurs, the signature request is not updated, remains in `WAIT_FOR_QTSP` and is retried later

**Review by gunzip — APPROVED** [review]

**Pull request merged**